src/projects/trio/trio.cpp

```diff
@@ -63,12 +66,38 @@ void HaplotypeRemover::deleteEdgeHaplo(int eid) {
     }
 }
 
+
+void HaplotypeRemover::compressAllVertices() {
+    size_t all_count = 0;
+    std::unordered_set<int> ids;
+    for (auto &v: mg.vertices) 
+        ids.insert(v.first);
+    for (auto id: ids) {
+        auto to_merge = mg.compressVertex(id);
+        for (auto p: to_merge){
+            HaplotypeStats new_haplo(haplotypes[p.second[0]]);
+            new_haplo.label = p.first;
+            haplotypes.insert(std::make_pair(p.first, new_haplo));
+            for (size_t i = 1; i < p.second.size(); i++) {
+                if (haplotypes[p.first].haplotype != haplotypes[p.second[i]].haplotype) {
+                    logger_.trace() << "Merging different haplotypes " << haplotypes[p.first].label <<
+                                    " " << haplotypes[p.second[i]].label << endl;
+                }
+                haplotypes[p.first].appendKmerStats(haplotypes[p.second[i]]);
+            }
+            all_count ++;
+        }
+    }
+    logger_.info() << "Compressing all " << all_count << endl;
+}
 void HaplotypeRemover::cleanGraph() {
     bool changed = true;
     size_t tips = 0;
     size_t bulges = 0;
+    size_t iter = 0;
     while (changed) {
         changed = false;
+        logger_.info() << "Iter " << ++iter << endl;
```

Comment:
Avoid things like "Iter" from info log. I will soon add a ProgressBar class to development that should be used in such cases.